Algorithms.Tests/Strings/PermutationTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Algorithms.Numeric;
+using Algorithms.Strings;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Strings
+{
+    public class AnagramTests
+    {
+
+        [TestCase("")]
+        [TestCase("A")]
+        [TestCase("abcd")]
+        [TestCase("aabcd")]
+        [TestCase("aabbbcd")]
+        [TestCase("aabbbccccd")]
+        public void Test_GetEveryUniquePermutation(string word)
+        {
+            var anagrams = Permutation.GetEveryUniquePermutation(word);
+
+            // make sure we have the right number of anagrams
+            var occDic = new Dictionary<char, int>();
```

Comment:
Please provide a better name, it's not clear what occ is and Dic is redundant if it means dictionary.